server/app/repository/PartRepository.ts

```diff
@@ -0,0 +1,47 @@
+import { injectable } from 'inversify';
+import { Schema } from 'mongoose';
+import { IPart, PartSchema } from './../models/PartModel';
+import { BaseRepository } from './BaseRepository';
+
+@injectable()
+export class PartRepository extends BaseRepository<IPart> {
+  protected readonly collectionName: string = 'parts';
+  protected readonly schema: Schema = PartSchema;
+
+  public async getByName(name: string): Promise<IPart | null> {
+    try {
+      return this.model.findOne({ name }).exec();
+    } catch (err) {
+      return this.manageRepositoryError(err);
+    }
+  }
+
+  public async updateByName(name: string, body: IPart): Promise<IPart | null> {
+    try {
+      return await this.model
+        .findOneAndUpdate({ name }, body, {
+          returnOriginal: false,
+          runValidators: true,
+        })
+        .exec();
+    } catch (err) {
+      return this.manageRepositoryError(err);
+    }
+  }
+
+  public async deletePart(name: string): Promise<IPart | null> {
+    try {
+      return await this.model.findOneAndDelete({ name }).then((deleteDocument) => {
+        if (deleteDocument) {
+          console.log(`Successfully deleted document: ${deleteDocument}`);
```

Comment:
should we potentially use the logger for where youre console.err or console.log

Reply:
Oh yeah definitely @SiuYe @Ming424